MSRV mismatch

Hello,

the README states that the MSRV is 1.59.0. However, when compiling for esp32 from main with the `1.61.0` toolchain, I get the following error:

```
   Compiling esp-hal-procmacros v0.1.0 (/var/home/ahartmann/repos/mt/playground/esp-hal/esp-hal-procmacros)
error[E0658]: use of unstable library feature 'bool_to_option'
   --> /var/home/ahartmann/repos/mt/playground/esp-hal/esp-hal-procmacros/src/lib.rs:228:35
    |
228 |         (f.sig.inputs.len() == 1).then_some(Ident::new("context", proc_macro2::Span::call_site()));
    |                                   ^^^^^^^^^
    |
    = note: see issue #80967 <https://github.com/rust-lang/rust/issues/80967> for more information
    = help: add `#![feature(bool_to_option)]` to the crate attributes to enable

For more information about this error, try `rustc --explain E0658`.
error: could not compile `esp-hal-procmacros` due to previous error
```

And indeed this language feature was added as part of [this PR][1] and landed in Rust 1.62.0. This bumps the MSRV to 1.62.0.

Maybe it would be a good idea to add some github action that compiles the code against the MSRV mentioned in the Readme so it doesn't break along the way, or at least gets recognized automatically?


[1]: https://github.com/rust-lang/rust/pull/96628

--- a/esp-hal-procmacros/src/lib.rs
+++ b/esp-hal-procmacros/src/lib.rs
@@ -225,7 +225,7 @@ pub fn interrupt(args: TokenStream, input: TokenStream) -> TokenStream {
     };
 
     let context_call =
-        (f.sig.inputs.len() == 1).then_some(Ident::new("context", proc_macro2::Span::call_site()));
+        (f.sig.inputs.len() == 1).then(|| Ident::new("context", proc_macro2::Span::call_site()));
 
     quote!(
         #(#cfgs)*

